webapp/src/components/History/History.tsx

```diff
@@ -1,5 +1,107 @@
-import React from 'react'
+import React, { ChangeEvent, useEffect, useState } from 'react'
+import { api } from '../../api'
+import {
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableFooter,
+  TableHead,
+  TablePagination,
+  TableRow,
+  Toolbar
+} from '@mui/material'
+import { PaymentEntry } from './PaymentEntry'
+import { useTranslation } from 'react-i18next'
 
 export const History: React.FC = () => {
-  return <p>History!</p>
+  const { t } = useTranslation('common')
+  const [history, setHistory] = useState<PaymentHistory>()
+
+  useEffect(() => {
+    queryHistory()
+  }, [])
+
+  const queryHistory = (page = 0, size = 10) => {
+    api.payment
+      .history({ page, size })
+      .then((data) => {
+        if (data.status === 200) {
+          setHistory(data.data)
+        }
+      })
+      .catch((err) => {
+        console.log(err)
+      })
+  }
+
+  const handleChangePage = (
+    event: React.MouseEvent<HTMLButtonElement, MouseEvent> | null,
+    newPage: number
+  ) => {
+    if (history) {
+      queryHistory(newPage, history?.pageable.pageSize)
+    }
+  }
+
+  const handleChangeRowsPerPage = (
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
+  ) => {
+    if (history) {
+      queryHistory(0, parseInt(event!.target.value))
+    }
+  }
+
+  return (
+    <div className="overflow-y-auto grow py-8 px-14 text-center">
+      <Toolbar />
```

Comment:
przeniesmy ten toolbar do UserLayout zeby go nie powtarzac
```html
<Sidebar />
<div>
    <Toolbar />
    <Outlet />
</div>
```

Reply:
na razie zostawie jak jest, nasz css to dramat troche i trzebaby zrobic taska zeby rozplanowac wszystko porzadnie, z uwzglednieniem responsywnosci. Po tej zaproponowanej zmianie rozjezdza sie troche strona np, w historii